pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -102,9 +101,31 @@ func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler netd
 		podInterfaceName: podInterfaceName,
 		iface:            correspondingNetworkIface,
 		launcherPID:      launcherPID,
+		dhcpConfigurator: dhcpConfigurator,
 	}, nil
 }
 
+func newBridgeDhcpConfigurator(iface v1.Interface, vmi v1.VirtualMachineInstance, podInterfaceName string, launcherPID *int, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) *dhcpconfigurator.DhcpConfigurator {
+	fakeBridgeIP, err := getFakeBridgeIP(iface.Name, vmi.Spec.Domain.Devices.Interfaces)
+	if err != nil {
+		return nil
+	}
+	fakeServerAddr, err := netlink.ParseAddr(fakeBridgeIP)
+	if err != nil || fakeServerAddr == nil {
+		return nil
+	}
+	bridgeIfaceName := fmt.Sprintf("k6t-%s", podInterfaceName)
+	dhcpServer := dhcpimpl.NewBridgeDhcpServer(bridgeIfaceName, *fakeServerAddr, handler, &vmi)
+	dhcpConfigurator := dhcpconfigurator.NewDhcpConfigurator(cacheFactory, getPIDString(launcherPID), dhcpServer)
+	return &dhcpConfigurator
+}
+
+func newMasqueradeDhcpConfigurator(podInterfaceName string, launcherPID *int, handler netdriver.NetworkHandler, cacheFactory cache.InterfaceCacheFactory) dhcpconfigurator.DhcpConfigurator {
```

Comment:
Let's return here `*DhcpConfigurator` like `newBridgeDhcpConfigurator` so the are used the sme way.

Reply:
Sure.